crates/uv-cache/src/removal.rs

```diff
@@ -40,7 +40,12 @@ impl Removal {
 
             // Remove the file.
             self.total_bytes += metadata.len();
-            remove_file(path)?;
+            if cfg!(windows) && metadata.is_symlink() {
+                // Remove the junction.
+                remove_dir(path)?;
+            } else {
+                remove_file(path)?;
+            }
```

Comment:
This is a bug. We have handling for this below (for entries), but not for the top-level thing we passed to `rm_rf`.